protocol/x/clob/types/process_proposer_matches_events.go

```diff
@@ -0,0 +1,56 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+)
+
+// ValidateProcessProposerMatchesEvents performs basic stateless validation on ProcessProposerMatchesEvents.
+// It returns an error if:
+// - Any of the fields have duplicate OrderIds.
+func (ppme *ProcessProposerMatchesEvents) ValidateProcessProposerMatchesEvents() error {
```

Comment:
If anyone has additional validation, lmk. I think there can be duplicates between the fields since we do all the filtering out (i.e filtering out placed, then cancelled stateful orders in the same block) when we read out PPME. 

In the future I think it would be nice to have specific update functions for each field so we don't have to validate all fields when we're only updating one subfield.

It would also be nice to do all the filtering out upon update instead of upon readout of PPME.